### PR TITLE
Header provider - rebased

### PR DESCRIPTION
**Pull request by dbernstein**
**This is @whikloj's PR (https://github.com/fcrepo4-exts/fcrepo4-vagrant/pull/92 - now closed) rebased and merged on my latest PR (https://github.com/fcrepo4-exts/fcrepo4-vagrant/pull/94).**
* * *
Once https://github.com/fcrepo4-exts/fcrepo4-vagrant/pull/94 has been merged,  I will merge this one.

Text of the original PR follows: 
***
**JIRA Ticket**: n/a

# What does this Pull Request do?
Installs a customized `fcrepo-config.xml` with the headerPrincipalProvider enabled.

# How should this be tested?

Need to stand up this vagrant, then create a resource and restrict it to one of the users (ie. adminuser). 

Then try to access that resource using the other user and the header
ie.
```
curl -i -u testuser:password1 -H"some-header: adminuser" http://localhost:8080/fcrepo/rest/your_restricted_resource
```
# Additional Notes:

Example:
* Does this change require documentation to be updated? yes, some information about the header might be helpful.
* Does this change add any new dependencies? no
* Does this change require any other modifications to be made to the repository (ie. Regeneration activity, etc.)? no
* Could this change impact execution of existing code? no

# Interested parties
@fcrepo4/committers

### CODE REVIEW

**Comment by dbernstein** [comment]
I have tested the header provider feature post rebase.  Everything works.  This PR is good to go barring any issues discovered by @yinlinchen on https://github.com/fcrepo4-exts/fcrepo4-vagrant/pull/94

File: README.md [hunk 1]
```diff
@@ -42,6 +42,12 @@ You can shell into the machine with `vagrant ssh` or `ssh -p 2222 vagrant@localh
     * Access via username = "karaf", password = "karaf"
     * Installed in karaf
 
+### Fedora Configuration
```
Comment (awoods) [inline README.md:45]: This section is duplicated below.

**Review by awoods — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (dbernstein) [reply README.md:45]: Fixed.

Comment (dbernstein) [reply README.md:45]: Fixed.

**Review by awoods — APPROVED** [review]

**Review by whikloj — APPROVED** [review]

**Pull request merged**